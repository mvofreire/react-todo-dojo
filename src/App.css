.App {
  text-align: center;
}

.link-logout {
  color: blue;
  text-decoration: underline;
  cursor: pointer;
}
/* # FORM */

.form-todo {
  position: relative;
  width: 965px;
  margin: auto;
  margin-bottom: 30px;
}

.error {
  font-size: 10px;
  color: red;
}

.input-form-todo {
  line-height: 30px;
  width: 460px;
  margin-right: 8px;
}

.btn {
  height: 30px;
}

/* # FORM*/

.todo-empty {
  color: #ddd;
  margin-top: 30px;
}

.lista {
  padding: 5px;
  width: 600px;
  margin: auto;
  display: flex;
  flex-flow: column;
}

.item-todo {
  width: 100%;
  line-height: 40px;
  margin: 2px 0px;
  padding: 0px 10px;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 0px 0px 0px 1px #ddd;
}

.item-todo:hover {
  background: #f3f3f3;
}

.item-todo input[type="checkbox"] {
  flex: 1;
}
.item-todo > span {
  flex: 12;
}
.item-todo button {
  flex: 1;
  text-align: center;
}

/*#region LOGIN*/

.login {
  background: #f1f1f1;
  width: 300px;
  margin: auto;
  padding: 30px 50px;
  border: solid 1px #ddd;
  border-radius: 4px;
  display: flex;
  flex-flow: column;
}
.login > input {
  line-height: 30px;
}
.login > .username {
  margin: 20px;
}
.login > .password {
  margin: 20px;
}

.login > .btn-login {
  margin: 20px;
}

.login > .login-error {
  margin: 20px;
  color: red;
}
/*#endregion*/
