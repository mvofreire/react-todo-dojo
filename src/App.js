import React, { Component } from "react";
import { TodoListApp, Login } from "./application";
import "./App.css";

class App extends Component {
  state = {
    user: false
  };

  componentDidMount() {
    this.validateLoggedUser();
  }

  validateLoggedUser = () => {
    if (localStorage && "user" in localStorage) {
      const user = JSON.parse(localStorage.getItem("user"));
      this.setUser(user);
    }
  };

  setUser = user => {
    this.setState({ user }, _ => {
      localStorage && localStorage.setItem("user", JSON.stringify(user));
    });
  };

  logout = () => {
    this.setUser(false);
  };

  render() {
    const { user } = this.state;
    return (
      <div className="App">
        <h1>TODO List App</h1>
        {user && (
          <span className="link-logout" onClick={this.logout}>
            logout ({user.nome})
          </span>
        )}
        {!!user ? <TodoListApp /> : <Login onAuthenticate={this.setUser} />}
      </div>
    );
  }
}

export default App;
