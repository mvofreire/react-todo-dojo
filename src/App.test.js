import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import localStorage from "./__mocks__/storage";

window.localStorage = localStorage;

describe("<App /> Component", () => {
  const user = { nome: "Teste" };

  beforeEach(() => localStorage.clear());

  it("renderiza sem erros", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renderiza login", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find("TodoListApp").length).toEqual(0);
    expect(wrapper.find("Login").length).toEqual(1);
  });

  it("valida localstorage quando recarrega", () => {
    localStorage.setItem("user", JSON.stringify(user));
    const wrapper = create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot("logado");
  });

  it("altera visualizacao apos login", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find("TodoListApp").length).toEqual(0);
    expect(wrapper.find("Login").length).toEqual(1);

    const instance = wrapper.instance();
    instance.setUser(user);

    expect(wrapper.find("TodoListApp").length).toEqual(1);
  });

  it("altera visualizacao apos deslogar", () => {
    const wrapper = create(<App />);
    wrapper.getInstance().logout();
    expect(wrapper.toJSON()).toMatchSnapshot("deslogado");
  });

  it("App Component to match snapshot", () => {
    const wrapper = create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot("deslogado");
  });

  it("App Component to match snapshot Logado", () => {
    const wrapper = create(<App />);
    const instance = wrapper.getInstance();
    instance.setUser(user);

    expect(wrapper.toJSON()).toMatchSnapshot("logado");
  });
});
