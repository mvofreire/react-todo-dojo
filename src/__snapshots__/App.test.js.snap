// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component App Component to match snapshot Logado: logado 1`] = `
<div
  className="App"
>
  <h1>
    TODO List App
  </h1>
  <span
    className="link-logout"
    onClick={[Function]}
  >
    logout (
    Teste
    )
  </span>
  <div
    className="App"
  >
    <div
      className="form-todo"
    >
      <br />
      <input
        className="input-form-todo"
        onChange={[Function]}
        onKeyUp={[Function]}
        value=""
      />
      <button
        className="btn btn-add"
        onClick={[Function]}
      >
        Adicionar
      </button>
      <button
        className="btn btn-clear"
        onClick={[Function]}
      >
        Limpar
      </button>
    </div>
    <div
      className="lista"
    />
    <span
      className="todo-empty"
    >
      Ainda nao foi cadastrado nenhum TODO
    </span>
  </div>
</div>
`;

exports[`<App /> Component App Component to match snapshot: deslogado 1`] = `
<div
  className="App"
>
  <h1>
    TODO List App
  </h1>
  <div
    className="login"
  >
    <input
      autoComplete="off"
      className="username"
      name="username"
      onChange={[Function]}
      placeholder="Usuario"
      type="text"
    />
    <input
      autoComplete="off"
      className="password"
      name="password"
      onChange={[Function]}
      placeholder="Senha"
      type="password"
    />
    <button
      className="btn btn-login"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;

exports[`<App /> Component altera visualizacao apos deslogar: deslogado 1`] = `
<div
  className="App"
>
  <h1>
    TODO List App
  </h1>
  <div
    className="login"
  >
    <input
      autoComplete="off"
      className="username"
      name="username"
      onChange={[Function]}
      placeholder="Usuario"
      type="text"
    />
    <input
      autoComplete="off"
      className="password"
      name="password"
      onChange={[Function]}
      placeholder="Senha"
      type="password"
    />
    <button
      className="btn btn-login"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;

exports[`<App /> Component valida localstorage quando recarrega: logado 1`] = `
<div
  className="App"
>
  <h1>
    TODO List App
  </h1>
  <div
    className="login"
  >
    <input
      autoComplete="off"
      className="username"
      name="username"
      onChange={[Function]}
      placeholder="Usuario"
      type="text"
    />
    <input
      autoComplete="off"
      className="password"
      name="password"
      onChange={[Function]}
      placeholder="Senha"
      type="password"
    />
    <button
      className="btn btn-login"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
